tensorflow/python/eager/backprop.py

```diff
@@ -923,6 +926,10 @@ def gradient(self,
     if self._tape is None:
       raise RuntimeError("GradientTape.gradient can only be called once on "
                          "non-persistent tapes.")
+    if not is_floating(self._tape):
```

Comment:
I don't think self._tape will ever be a floating point tensor.

Reply:
Note that a similar check in `gradient` is already covered by #27183.